native/src/modules/endpoint/reducers/refreshCategory.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import type { CategoryRouteStateType, CityContentStateType, PathType } from '../../app/StateType'
+import type { PushCategoryActionType } from '../../app/StoreActionType'
+import { CategoryModel, LanguageModel } from '@integreat-app/integreat-api-client'
+import forEachTreeNode from '../../common/forEachTreeNode'
+import ErrorCodes from '../../error/ErrorCodes'
+
+const getAllAvailableLanguages = (
+  category: CategoryModel, city: string, language: string, cityLanguages: Array<LanguageModel>
+) => {
+  if (category.isRoot()) {
+    return new Map(cityLanguages.map(model => [model.code, `/${city}/${model.code}`]))
+  }
+  const allAvailableLanguages = new Map(category.availableLanguages)
+  allAvailableLanguages.set(language, category.path)
+  return allAvailableLanguages
+}
```

Comment:
Is this only used here and in `pushCategory`? Maybe this could be extracted to a separate function. If only used in these to places probably better to keep it the current way.

Reply:
As far as I can tell it's only used in those two places.